src/main/scala/lc/Main.scala

```diff
@@ -86,6 +90,17 @@ class Captcha(throttle: Int) {
     val thread = ex.scheduleWithFixedDelay(task, 1, delay, TimeUnit.SECONDS)
   }
 
+  def getChallenge(param: Parameters): Id = {
+    val rs = stmt.executeQuery("SELECT token FROM challenge WHERE solved=FALSE LIMIT 1")
+    var id: String = null
+    if(rs.next()){
+      id = rs.getString("token")
+    } else {
+      id = generateChallenge(param)
+    }
```

Comment:
In scala, an if..else statement is also an expression, so you can rewrite the above as:

```scala
val id = if (rs.next) {
  rs.getString("token")
} else {
  generateChallenge(param)
}
```

Reply:
This works.